Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/JobUtils.java

```diff
@@ -675,6 +676,20 @@ public static String uploadFileToCluster(@NotNull final IClusterDetail selectedC
         });
     }
 
+    public static Observable<String> deployArtifact(@NotNull  SparkBatchSubmission submission, @NotNull String clusterName,@NotNull String webHdfsUploadPath, @NotNull  String artifactPath) {
```

Comment:
You have put `UploadByWebhdfs` to Spark submission, I suppose the method can be put into SparkBatchJob